### PR TITLE
chore: update Node to 20 LTS, TypeScript, tsconfig

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* We want to update to Node 20 LTS.
* We want to target the JS output of Node 20.
* We want to use the latest TypeScript 5.4

This commit:

* Updates the Node version to 20 LTS.
* Updates the TypeScript version and related TS libs
  to 5.4.
* Updates tsconfig for the new Node 20 target.

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.

### CODE REVIEW

File: packages/fxa-admin-panel/package.json [hunk 1]
```diff
@@ -84,7 +84,7 @@
     "prettier": "^2.3.1",
     "supertest": "^6.3.0",
     "tailwindcss-textshadow": "^2.1.3",
-    "typescript": "^5.3.3",
+    "typescript": "^5.4.2",
```
Comment (julianpoy) [inline packages/fxa-admin-panel/package.json:87]: Can this and the other package-based references be removed?

**Review by julianpoy — APPROVED** [review]

File: packages/fxa-admin-panel/package.json (hunk 1 above)
Comment (StaberindeZA) [reply packages/fxa-admin-panel/package.json:87]: Unfortunately at the moment no. See this comment from PR #16390.

> Readd a few dev dependancies back into packages/**/package.json
>
>  Behavior of npx nx... commands seems to have changed, resulting in
>  multiple "command not found: " errors, where
>  refers to libaries used in packages/**/package.json scripts, such as
>  tailwindcss, tsc-alias, nx, etc.

File: packages/fxa-graphql-api/src/gql/dto/input/session-reauth.ts [hunk 2]
```diff
@@ -14,5 +14,5 @@ export class SessionReauthInput extends SignInInput {
   public sessionToken!: hexstring;
 
   @Field()
-  public options!: SessionReauthOptionsInput;
+  public declare options: SessionReauthOptionsInput;
```
Comment (dschom) [inline packages/fxa-graphql-api/src/gql/dto/input/session-reauth.ts:17]: Should this be `options!` instead of `options`?

Comment (bbangert) [reply packages/fxa-graphql-api/src/gql/dto/input/session-reauth.ts:17]: Nope, that will result in an error as `declare` doesn't need it.

Comment (dschom) [reply packages/fxa-graphql-api/src/gql/dto/input/session-reauth.ts:17]: Gotcha, just realized that.

Comment (dschom) [reply packages/fxa-graphql-api/src/gql/dto/input/session-reauth.ts:17]: hmmm... The SessionReauthOptionsInput is actually the same as the base classes options type... maybe this isn't needed at all... Probably good for separate commit though.

File: package.json [hunk 3]
```diff
@@ -108,12 +108,12 @@
...
     "rxjs": "^7.8.1",
     "semver": "^7.6.0",
-    "tslib": "^2.5.0",
+    "tslib": "^2.6.2",
     "uuid": "^9.0.0",
     "winston": "^3.11.0"
   },
   "engines": {
-    "node": "^18.18.2"
+    "node": "^20.11.1"
```
Comment (dschom) [inline package.json:116]: This is referenced in pm2-all.sh to stop startup if the system's current node version doesn't match. It appears this check isn't running though... maybe this is something for a follow up? We should validate when running `yarn start` and it'd be nice to validate in this in the CI as well. I wouldn't block on this though.

**Review by dschom — APPROVED** [review]
Looks good! See optional comment on package.json.

**Pull request merged**